core/embed/rust/src/trezorhal/uzlib.rs

```diff
@@ -0,0 +1,42 @@
+use super::ffi;
+use core::ptr;
```

Comment:
rustification done in 8e644a6641cb515073e2c43227c33b720ff3b263, though its a bit different as window usage is optional and dest_buf can have arbitrary size

Reply:
any reason `dest_buf` is passed at initialization time instead into the decompression function?

Reply:
do we know what is the purpose of not specifying a window?

Reply:
dest_buf: saves a bit of time by only calculating dest_limit only once maybe? would there by any benefit in using different buffers when iterating?

window: i don't really understand what is happening inside the uzlib, but it seems like the window is not needed when doing full-dest-in-memory

Reply:
one clear advantage of passing `dest_buf` directly to decompressin function is that i can use the data in dest_buf directly instead of copying the return value, which comes handy when using larger buffer that 1 byte. so changed this in https://github.com/trezor/trezor-firmware/commit/a173efb3834e1c28c28aede9e53967b58a0d1b84 and also added some reaction to uzlib error